lib/jit_preloader/active_record/base.rb

```diff
@@ -41,11 +41,22 @@ def has_many_aggregate(assoc, name, aggregate, field, default: 0)
             association_scope = klass.all.merge(association(assoc).scope).unscope(where: aggregate_association.foreign_key)
             association_scope = association_scope.instance_exec(&reflection.scope).reorder(nil) if reflection.scope
 
-            conditions[aggregate_association.table_name] = { aggregate_association.foreign_key => primary_ids }
+            # If the query uses an alias for the association, use that instead of the table name
+            table_alias_name = association_scope.references_values.first || aggregate_association.table_name
+            conditions[table_alias_name] = { aggregate_association.foreign_key => primary_ids }
+
+            # If the association is a STI child model, specify its type in the condition so that it
+            # doesn't include results from other child models
+            is_base_class = aggregate_association.klass.superclass.abstract_class? || aggregate_association.klass.superclass == ActiveRecord::Base
+            has_type_column = aggregate_association.klass.column_names.include?(aggregate_association.klass.inheritance_column)
```

Comment:
I thinks all of this could be replaced with
```
if aggregate_association.klass != aggregate_association.klass.base_class
  # At this point I believe you know that `aggregate_association.klass` is an STI table and itsn't the base class
end
```

Reply:
Not necessarily. You can have one ActiveRecord model inherit from another without using STI. I'm not sure what the situation is where you would want to do this, but it's probably better to be safe here.

I was able to duplicate this by doing the following:
- Make a `PostalAddress` model that inherited from the `Address` model
- Update all of the `addresses` references on the other models to `postal_addresses`
- Update the `Contact` instances generated in the specs to use the `PostalAddress` model
- Add a debugger before the `if` statement
- Run this test https://github.com/clio/jit_preloader/blob/fe7cc4c543f29d04d9c61ca53556846ba7c3e029/spec/lib/jit_preloader/preloader_spec.rb#L276

Reply:
Does this only cover the case where our class hierarchy is:
```
ActiveRecord::Base
  GenericParent
     ChildSTIModel
     ChildSTIModel2
```

What if there was another parent in between? Is this something we need to consider?

Reply:
I believe the logic we have should cover cases where there is a parent in between. The logic is checking two things:
- That the direct parent of the class isn't an abstract class or `ActiveRecord::Base`
- That the class has the inheritance column defined on its DB table (which is how it identifies it as an STI model)

This logic should correctly identify an STI model in the case that you defined. That said, I think any child of an ActiveRecord model that has defined the inheritance column is an STI model.